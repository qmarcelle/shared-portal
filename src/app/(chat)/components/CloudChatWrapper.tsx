'use client';

// CloudChatWrapper injects the Genesys Cloud Messenger SDK and manages Messenger-specific events.
// All script loading, Messenger events, and errors are logged for traceability and debugging.

import { useEffect } from 'react';
import { useChatStore } from '../stores/chatStore';

export default function CloudChatWrapper() {
  const { userData } = useChatStore();

  useEffect(() => {
    // Build chatSettings from env and userData
    window.chatSettings = {
      bootstrapUrl: process.env.NEXT_PUBLIC_GENESYS_BOOTSTRAP_URL || '',
      widgetUrl: process.env.NEXT_PUBLIC_GENESYS_WIDGET_URL || '',
      clickToChatJs: process.env.NEXT_PUBLIC_GENESYS_CLICK_TO_CHAT_JS || '',
      clickToChatEndpoint: process.env.NEXT_PUBLIC_CLICK_TO_CHAT_ENDPOINT || '',
      chatTokenEndpoint: process.env.NEXT_PUBLIC_CHAT_TOKEN_ENDPOINT || '',
      coBrowseEndpoint: process.env.NEXT_PUBLIC_COBROWSE_LICENSE_ENDPOINT || '',
      opsPhone: process.env.NEXT_PUBLIC_OPS_PHONE || '',
      opsPhoneHours: process.env.NEXT_PUBLIC_OPS_HOURS || '',
      ...userData,
    };
    console.debug('[CloudChatWrapper] chatSettings:', window.chatSettings);
  }, [userData]);

  useEffect(() => {
    // Load Genesys Cloud bootstrap and widget scripts
    const bootstrapUrl = window.chatSettings?.bootstrapUrl || '';
    const widgetUrl = window.chatSettings?.widgetUrl || '';
    if (!bootstrapUrl) return;
    const bootstrapScript = document.createElement('script');
    bootstrapScript.src = bootstrapUrl;
    bootstrapScript.async = true;
    bootstrapScript.onload = () => {
      console.debug('[CloudChatWrapper] Genesys bootstrap loaded');
      if (!widgetUrl) return;
      const widgetScript = document.createElement('script');
      widgetScript.src = widgetUrl;
      widgetScript.async = true;
      widgetScript.onload = () => {
        console.debug('[CloudChatWrapper] Genesys widgets loaded');
      };
      widgetScript.onerror = (e) => {
        console.error('[CloudChatWrapper] Failed to load widgets script', e);
      };
      document.body.appendChild(widgetScript);
    };
    bootstrapScript.onerror = (e) => {
      console.error('[CloudChatWrapper] Failed to load bootstrap script', e);
    };
    document.body.appendChild(bootstrapScript);

    // Provide openGenesysChat logic
    window.openGenesysChat = () => {
      if (window.Genesys) {
        try {
          window.Genesys('command', 'Messenger.open');
        } catch (e) {
          console.error(
            '[CloudChatWrapper] Error opening Genesys Messenger',
            e,
          );
        }
      }
    };

    return () => {
      // Cleanup scripts and chatSettings
      document.body.removeChild(bootstrapScript);
      if (widgetUrl) {
        const widgetScript = document.querySelector(
          'script[src="' + widgetUrl + '"]',
        );
        if (widgetScript) document.body.removeChild(widgetScript);
      }
      if (window.chatSettings) delete window.chatSettings;
      if (window.openGenesysChat) delete window.openGenesysChat;
      console.debug('[CloudChatWrapper] Cleanup complete');
    };
  }, [userData]);

  return null;
}
