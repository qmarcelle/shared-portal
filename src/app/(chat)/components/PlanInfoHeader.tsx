'use client';
import { useChatStore } from '@/app/(chat)/stores/chatStore';
import { Title } from '@/components/foundation/Title';

/**
 * Show current plan when multiple plans exist
 */
export default function PlanInfoHeader() {
  const { formInputs } = useChatStore();
  if (formInputs.length <= 1) return null;
  const plan = formInputs.find((f) => f.id === 'PLAN_ID');

  return (
    <Title
      className="plan-info-header"
      text={`You are chatting about plan ${plan?.value || ''}`}
    />
  );
}
