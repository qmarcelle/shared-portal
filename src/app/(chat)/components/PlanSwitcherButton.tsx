'use client';
import { useChatStore } from '@/app/(chat)/stores/chatStore';
import { Button } from '@/components/foundation/Button';
import { usePlanStore } from '../../../userManagement/stores/planStore';

/**
 * "Switch Plan" inside pre-chat window
 */
export default function PlanSwitcherButton() {
  const { endChat } = useChatStore();
  const { plans, selectedPlanId, openPlanSwitcher } = usePlanStore();

  if (plans.length <= 1) return null;

  const handleSwitch = () => {
    endChat();
    openPlanSwitcher();
  };

  return (
    <Button
      label="Switch Plan"
      callback={handleSwitch}
      className="plan-switch-button"
      type="secondary"
    />
  );
}
