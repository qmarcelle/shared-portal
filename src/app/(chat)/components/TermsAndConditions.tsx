'use client';
// import TCs from '@/app/@chat/config/terms'; // { default: '…', lobA: '…', … }
import { useChatStore } from '@/app/(chat)/stores/chatStore';
import { RichText } from '@/components/foundation/RichText';

/**
 * Render T&C based on LOB
 */
export default function TermsAndConditions() {
  const { userData } = useChatStore();
  // const text = TCs[userData.LOB] || TCs.default;
  return (
    <RichText
      spans={['Terms and Conditions']}
      className="terms-and-conditions"
      type="body-2"
    />
  );
}
