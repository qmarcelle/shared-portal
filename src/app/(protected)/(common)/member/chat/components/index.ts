/**
 * Component Exports
 * Centralizes all component exports following the new consolidated structure
 */

// Main Components
export { ChatTrigger } from './ChatTrigger';
export { ChatWidget } from './ChatWidget';
export { PlanInfoHeader } from './PlanInfoHeader';

// Error Handling
export { ChatErrorBoundary } from './shared/ChatErrorBoundary';

// Integration Components (to be consolidated)
export { GenesysScripts } from './GenesysScripts';
