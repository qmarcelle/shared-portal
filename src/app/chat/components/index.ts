// Core Components
export * from './core/ChatButton';
export * from './core/ChatErrorBoundary';
export * from './core/ChatHeader';
export * from './core/ChatWidget';

// Business Hours Components
export * from './business-hours/BusinessHoursNotification';

// Eligibility Components
export * from './eligibility/EligibilityCheck';

// Plan Switcher Components will be added here when implemented
// export * from './plan-switcher/PlanSwitcher';
