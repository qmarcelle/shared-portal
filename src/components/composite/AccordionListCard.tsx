import { AccordionInfoItem } from '@/models/accordionInfoItem';
import { IComponent } from '../IComponent';
import { Accordion } from '../foundation/Accordion';
import { Card } from '../foundation/Card';
import { Column } from '../foundation/Column';
import { Divider } from '../foundation/Divider';
import { Header } from '../foundation/Header';
import { Spacer } from '../foundation/Spacer';

interface AccordionListCardProps extends IComponent {
  header: string;
  information: AccordionInfoItem[];
}

export const AccordionListCard = ({
  header,
  information,
  className,
}: AccordionListCardProps) => {
  return (
    <Card className={`${className ?? ''} ${header ? 'large-section' : ''}`}>
      <Column className="items-stretch">
        {header && (
          <div>
            <Header className="title-2" text={header}></Header>
            <Spacer size={22} />
          </div>
        )}
        {information.map((info, index) => (
          <>
            <Accordion
              key={index}
              className="px-2 py-4 "
              label={info.title}
              icon={info.icon}
              child={info.body}
              initialOpen={false}
              type="card"
              openIcon={<Image className="pl-2 w-6" src={Down} alt=""></Image>}
              closeIcon={<Image className="pl-2 w-6" src={Up} alt=""></Image>}
              onOpenCallBack={() =>
                info.onOpenCallBack && info.onOpenCallBack()
              }
            ></Accordion>
            {index !== information.length - 1 && <Divider />}
          </>
        ))}
      </Column>
    </Card>
  );
};
