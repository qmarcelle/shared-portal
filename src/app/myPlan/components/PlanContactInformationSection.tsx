import { IComponent } from '@/components/IComponent';
import { AppLink } from '@/components/foundation/AppLink';
import { Column } from '@/components/foundation/Column';
import { Spacer } from '@/components/foundation/Spacer';
import { TextBox } from '@/components/foundation/TextBox';

interface PlanContactInformationSectionProps extends IComponent {
  title: string;
  address: string;
  primaryPhoneNumber: string;
  secondaryPhoneNumber: string;
}

export const PlanContactInformationSection = ({
  title,
  address,
  primaryPhoneNumber,
  secondaryPhoneNumber,
}: PlanContactInformationSectionProps) => {
  return (
    <Column className="flex flex-col">
      <Spacer size={16} />
      <TextBox className="body-1" text={title}></TextBox>
      <Spacer size={32} />
      <TextBox className="body-2" text="Mailing Address"></TextBox>
      <TextBox className="body-1" text={address}></TextBox>
      <Spacer size={16} />
      <TextBox className="body-2" text="Primary Phone Number"></TextBox>
      <TextBox className="body-1" text={primaryPhoneNumber}></TextBox>
      <Spacer size={16} />
      <TextBox className="body-2" text="Secondary Phone Number"></TextBox>
      <TextBox className="body-1" text={secondaryPhoneNumber}></TextBox>
      <Spacer size={32} />
      <AppLink
        label="View All Plan Contact Information"
        url="/member/myplan/plancontact"
      />
    </Column>
  );
};
