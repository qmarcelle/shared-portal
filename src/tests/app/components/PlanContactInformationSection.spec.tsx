import { AccordionListCard } from '@/components/composite/AccordionListCard';
import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import { PlanContactInformationSection } from '../../../app/myPlan/components/PlanContactInformationSection';

const renderUI = () => {
  return render(
    <AccordionListCard
      header=""
      information={[
        {
          title: 'View Plan Contact Information',
          body: (
            <PlanContactInformationSection
              title="Below is the phone number and mailing address associated with your plan."
              address="123 Street Address Road City Town, TN 12345"
              primaryPhoneNumber="(123) 456-7890"
              secondaryPhoneNumber="NA"
            />
          ),
        },
      ]}
    ></AccordionListCard>,
  );
};

describe('PlanContactInformation', () => {
  it('should render UI correctly', () => {
    const component = renderUI();

    screen.getByText('View Plan Contact Information');
    let planContactInformationContent = screen.queryByText(
      '123 Street Address Road City Town, TN 12345',
    );
    expect(planContactInformationContent).not.toBeInTheDocument();

    fireEvent.click(screen.getByText('View Plan Contact Information')); //check that body text is visible after clicking
    planContactInformationContent = screen.getByText(
      '123 Street Address Road City Town, TN 12345',
    );
    expect(planContactInformationContent).toBeInTheDocument();

    fireEvent.click(screen.getByText('View Plan Contact Information')); //check that it is not visible after clicking again
    planContactInformationContent = screen.queryByText(
      '123 Street Address Road City Town, TN 12345',
    );
    expect(planContactInformationContent).not.toBeInTheDocument();

    expect(component.baseElement).toMatchSnapshot();
  });
});
