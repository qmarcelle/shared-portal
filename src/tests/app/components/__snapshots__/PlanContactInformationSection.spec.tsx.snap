// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanContactInformation should render UI correctly 1`] = `
<body>
  <div>
    <div
      class="card-main "
    >
      <div
        class="flex flex-col items-stretch"
      >
        <div
          class="flex flex-col card-main-elevated p-1 cursor-pointer px-2 py-4"
        >
          <div
            class="flex flex-row m-1 accordion-head"
          >
            <p
              class="mr-2 flex-grow"
            >
              View Plan Contact Information
            </p>
            <img
              alt="Down Chevron"
              class="pl-2 w-6"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
          </div>
          <div
            class="mr-2 flex-grow"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
